src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc

```diff
@@ -0,0 +1,289 @@
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = "valid_navigation_text";
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "invalid_turn_icon\t\n";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest,
+       Run_ProcessSoftButtonsNotSucceeded_UNSUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = "valid_navigation_text";
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "valid_turn_icon";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  const mobile_result::eType kExpecterResult = mobile_result::INVALID_ENUM;
+  EXPECT_CALL(*mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(kExpecterResult));
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(MobileResultCodeIs(kExpecterResult), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest, Run_NoTurnList_UNSUCCESS) {
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  EXPECT_CALL(*mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(mobile_result::SUCCESS));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest, Run_SUCCESS) {
+  const std::string kNavigationText = "valid_navigation_text";
+
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = kNavigationText;
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "valid_turn_icon";
+  (*command_msg_)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@LevchenkoS Do you really dereference a NULL pointer? If so this means undefined behavior!